### PR TITLE
Update CHANGELOG with 2.10.0 changes

### PR DESCRIPTION
**Pull request by stepankuzmin**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,26 @@
+## 2.10.0
+
+## Features ✨ and improvements 🏁
+* Adding new marker styling option: rotationAlignment: 'horizon' allowing marker rotation to match the curvature of the horizon in globe view ([#11894](https://github.com/mapbox/mapbox-gl-js/pull/11894))
```
Comment (SnailBones) [inline CHANGELOG.md:4]:
```suggestion
* Add new marker styling option `rotationAlignment: 'horizon'` allowing marker rotation to match the curvature of the horizon in globe view ([#11894](https://github.com/mapbox/mapbox-gl-js/pull/11894))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+* Improve globe-mercator transition and map load performance with globe projection ([#12039](https://github.com/mapbox/mapbox-gl-js/pull/12039))
+* Improve panning precision on Globe View and relax constrains on lower zoom levels ([#12114](https://github.com/mapbox/mapbox-gl-js/pull/12114))
+
+
+## Bug fixes 🐞
+* Fix potential flickering on on image source updates ([#11928](https://github.com/mapbox/mapbox-gl-js/pull/11928))
+* Copy stylesheet to allow toggling different styles using setStyle without overwriting some of the properties ([#11942](https://github.com/mapbox/mapbox-gl-js/pull/11942))
+* Fix a bug where id expression didn't work for 0 values ([#12000](https://github.com/mapbox/mapbox-gl-js/pull/12000))
+* Fix precision errors in depth pack/unpack ([#12005](https://github.com/mapbox/mapbox-gl-js/pull/12005))
+* Allow style spec to use fog default values when toggling styles  ([#11992](https://github.com/mapbox/mapbox-gl-js/pull/11992))
```
Comment (avpeery) [inline CHANGELOG.md:16]: Double checking here but I thought 11992 was in the 2.9.0 release

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+* Fix occasional missing tiles at bottom of screen during globe-Mercator transition ([#11988](https://github.com/mapbox/mapbox-gl-js/pull/11988))
```
Comment (SnailBones) [inline CHANGELOG.md:17]: This was reverted by https://github.com/mapbox/mapbox-gl-js/pull/12075 and will be fixed when https://github.com/mapbox/mapbox-gl-js/pull/12137 is merged
```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,26 @@
...
+* Fix `cooperativeGestures` preventing panning on mobile while in fullscreen. Add screen reader alert for cooperative gestures warning message. ([#12058](https://github.com/mapbox/mapbox-gl-js/pull/12058))
```
Comment (SnailBones) [inline CHANGELOG.md:18]: Perhaps we should consider listing these related but distinct changes separately?
```suggestion
* Fix `cooperativeGestures` preventing panning on mobile while in fullscreen. ([#12058](https://github.com/mapbox/mapbox-gl-js/pull/12058))
* Add screen reader alert for cooperative gestures warning message. ([#12058](https://github.com/mapbox/mapbox-gl-js/pull/12058))
```

**Review by SnailBones — APPROVED** [review]
Added a few nits, otherwise LGTM!

File: CHANGELOG.md (hunk 2 above)
Comment (avpeery) [inline CHANGELOG.md:16]: I'm pretty sure the copy the stylesheet to allow toggling setStyle and fog default values when toggling setStyle were in the 2.9.0 release. This release only includes this setStyle fix here: https://github.com/mapbox/mapbox-gl-js/pull/12049

Comment (avpeery) [reply CHANGELOG.md:16]: Confirmed #11992 and #11942 were in 2.9.0 so they can get removed from the changelog. Included in this cherry pick: https://github.com/mapbox/mapbox-gl-js/pull/11979

Comment (SnailBones) [reply CHANGELOG.md:16]: Good catch @avpeery! This appears in the changelog [below ](https://github.com/mapbox/mapbox-gl-js/blob/main/CHANGELOG.md?plain=1#L30) for 2.9.0.
```suggestion
```

**Review by avpeery — CHANGES_REQUESTED** [review]
See comment above about some entries being included already in the 2.9.0 release. I'll check for any others!

File: CHANGELOG.md (hunk 1 above)
Comment (avpeery) [reply CHANGELOG.md:4]: This PR was also already introduced in v2.9.0 see: https://github.com/mapbox/mapbox-gl-js/pull/11946

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,26 @@
...
+* Adding new marker styling option: rotationAlignment: 'horizon' allowing marker rotation to match the curvature of the horizon in globe view ([#11894](https://github.com/mapbox/mapbox-gl-js/pull/11894))
+* Improve rendering performance on globe view ([#12050](https://github.com/mapbox/mapbox-gl-js/pull/12050))
+* Improve tile loading performance on low zoom levels ([#12061](https://github.com/mapbox/mapbox-gl-js/pull/12061))
+* Improve globe-mercator transition and map load performance with globe projection ([#12039](https://github.com/mapbox/mapbox-gl-js/pull/12039))
+* Improve panning precision on Globe View and relax constrains on lower zoom levels ([#12114](https://github.com/mapbox/mapbox-gl-js/pull/12114))
+
+
+## Bug fixes 🐞
+* Fix potential flickering on on image source updates ([#11928](https://github.com/mapbox/mapbox-gl-js/pull/11928))
+* Copy stylesheet to allow toggling different styles using setStyle without overwriting some of the properties ([#11942](https://github.com/mapbox/mapbox-gl-js/pull/11942))
```
Comment (avpeery) [inline CHANGELOG.md:13]: As mentioned #11942 was introduced in 2.9.0 release. We'll have to check out the github release tools to see why it let some of these pass through!

Comment (SnailBones) [reply CHANGELOG.md:13]: My guess is that GH release tools doesn't account for cherry-picking. But that would be great feature to add!

```suggestion
```

**Comment by stepankuzmin** [comment]
Thanks, @SnailBones @avpeery! 🙇

**Comment by avpeery** [comment]
Fixed a small typo, looks good to me! thanks @stepankuzmin

**Review by avpeery — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,22 @@
...
+* Improve globe-mercator transition and map load performance with globe projection ([#12039](https://github.com/mapbox/mapbox-gl-js/pull/12039))
+* Improve panning precision on Globe View and relax constraints on lower zoom levels ([#12114](https://github.com/mapbox/mapbox-gl-js/pull/12114))
+
+
+## Bug fixes 🐞
+* Fix a bug where id expression didn't work for 0 values ([#12000](https://github.com/mapbox/mapbox-gl-js/pull/12000))
+* Fix precision errors in depth pack/unpack ([#12005](https://github.com/mapbox/mapbox-gl-js/pull/12005))
+* Fix `cooperativeGestures` preventing panning on mobile while in fullscreen. ([#12058](https://github.com/mapbox/mapbox-gl-js/pull/12058))
+* Add screen reader alert for cooperative gestures warning message. ([#12058](https://github.com/mapbox/mapbox-gl-js/pull/12058))
+* Fix clearing raster tiles when toggling "setStyle" with a globe projection ([#12049](https://github.com/mapbox/mapbox-gl-js/pull/12049))
```
Comment (SnailBones) [inline CHANGELOG.md:15]: Nit: formatting
```suggestion
* Fix clearing raster tiles when toggling `setStyle` with a globe projection ([#12049](https://github.com/mapbox/mapbox-gl-js/pull/12049))
```

**Pull request merged**